src/main/java/net/datafaker/transformations/JsonTransformer.java

```diff
@@ -7,149 +7,185 @@
 import java.util.*;
 
 public class JsonTransformer<IN> implements Transformer<IN, Object> {
-  private static final Map<Character, String> ESCAPING_MAP = createEscapeMap();
-
-  @Override
-  public String apply(IN input, Schema<IN, ?> schema) {
-    Field<?, ?>[] fields = schema.getFields();
-    StringBuilder sb = new StringBuilder();
-    sb.append("{");
-    for (int i = 0; i < fields.length; i++) {
-      value2String((fields[i].getName()), sb);
-      sb.append(": ");
-      if (fields[i] instanceof CompositeField) {
-        sb.append(apply(input, (CompositeField) fields[i], i));
-      } else {
-        value2String(((SimpleField) fields[i]).transform(input), sb);
-      }
-      if (i < fields.length - 1) {
-        sb.append(", ");
-      }
+
+    private static final Map<Character, String> ESCAPING_MAP = createEscapeMap();
+
+    private boolean formatAsArray;
+
+    private JsonTransformer() {
+        this(false);
     }
-    sb.append("}");
-    return sb.toString();
-  }
-
-  @Override
-  public String generate(FakeSequence<IN> input, Schema<IN, ?> schema) {
-    if (input.isInfinite()) {
-      throw new IllegalArgumentException("The sequence should be finite of size");
+
+    private JsonTransformer(boolean asArray) {
+        this.formatAsArray = asArray;
     }
 
-    StringJoiner data = new StringJoiner(LINE_SEPARATOR);
-    for (IN in : input) {
-      data.add(apply(in, schema));
+    @Override
+    public String apply(IN input, Schema<IN, ?> schema) {
+        Field<?, ?>[] fields = schema.getFields();
+        StringBuilder sb = new StringBuilder();
+        sb.append("{");
+        for (int i = 0; i < fields.length; i++) {
+            value2String((fields[i].getName()), sb);
+            sb.append(": ");
+            if (fields[i] instanceof CompositeField) {
+                sb.append(apply(input, (CompositeField) fields[i], i));
+            } else {
+                value2String(((SimpleField) fields[i]).transform(input), sb);
+            }
+            if (i < fields.length - 1) {
+                sb.append(", ");
+            }
+        }
+        sb.append("}");
+        return sb.toString();
     }
```

Comment:
Why do we need to touch these lines within this PR?

Reply:
editorconfig violation? formatting only. Not sure why this file seemed to have drifted from others - i did check the history and it's always had 2 indent. 🤷

Reply:
it's better not to mix checkstyle changes with PR related in same commit

Reply:
I use intellij. It bundles and EditorConfig plugin and applies formatting changes automatically.